MonoGame.Framework.Content.Pipeline/ContextScopeFactory.cs

```diff
@@ -0,0 +1,209 @@
+// MonoGame - Copyright (C) MonoGame Foundation, Inc
+// This file is subject to the terms and conditions defined in
+// file 'LICENSE.txt', which is part of this source code package.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using Microsoft.Xna.Framework.Content.Pipeline;
+using MonoGame.Framework.Content.Pipeline.Builder;
+
+namespace MonoGame.Framework.Content
+{
+
+    /// <summary>
+    /// The <see cref="IContentContext"/> represents some sort of context operation's context.
+    /// Usually, this represents a <see cref="ContentImporterContext"/>
+    /// or <see cref="ContentProcessorContext"/>.
+    ///
+    /// However, because those types are part of the XNA namespace, they cannot be modified directly.
+    /// This interface is an adapter over those types.
+    /// </summary>
+    public interface IContentContext : IDisposable
+    {
+        /// <inheritdoc cref="ContentImporterContext.IntermediateDirectory"/>
+        public string IntermediateDirectory { get; }
+
+        /// <inheritdoc cref="ContentImporterContext.Logger"/>
+        public ContentBuildLogger Logger { get; }
+
+        /// <inheritdoc cref="ContentProcessorContext.SourceIdentity"/>
+        public ContentIdentity SourceIdentity { get; }
+    }
+
+    /// <summary>
+    /// <para>
+    /// The <see cref="ContextScopeFactory"/> facilitates access to a <see cref="IContentContext"/>
+    /// instance without direct access to the actual context.
+    /// </para>
+    ///
+    /// <para>
+    /// Anytime a content context operation is about to start, the operation should signal
+    /// the <see cref="BeginContext(IContentContext)"/> method.
+    /// <b> Critically </b>, the content operation must <b>dispose</b> the resulting context
+    /// when the operation has concluded.
+    /// </para>
+    ///
+    /// <para>
+    /// The most recent content operation can be retrieved with the <see cref="ActiveContext"/>
+    /// property.
+    /// </para>
+    ///
+    /// </summary>
+    public static class ContextScopeFactory
```

Comment:
does this need to be public or internal (I know the tests need it , but we can use InternalsVisibleTo as we do for others) same for the `interface`